components/authz-service/server/v2/policy.go

```diff
@@ -859,8 +869,18 @@ func statementFromAPI(statement *api.Statement) (storage.Statement, error) {
 		statement.Resources = []string{"*"}
 	}
 
-	// map external representation of "all projects" to actual ID for that meta-project
 	projects := make([]string, len(statement.Projects))
+
+	if len(statement.Projects) == 0 {
+		if s.isBeta2p1() {
```

Comment:
Everything above here was to be able to have this function exist--now we can distinguish v2/v2.1 and take appropriate action.